test/Microsoft.AspNetCore.Server.Kestrel.Performance/ResponseHeaders.cs

```diff
@@ -0,0 +1,215 @@
+            response.Headers["Content-Range"] = "Test Value";
+            response.Headers["Expires"] = "Test Value";
+            response.Headers["Last-Modified"] = "Test Value";
+            var dateHeaderValues = DateHeaderValueManager.GetDateHeaderValues();
+            HeadersDirect.SetRawDate(dateHeaderValues.String, dateHeaderValues.Bytes);
+            return Task.CompletedTask;
+        };
+
+        private static readonly RequestDelegate Unknown = (context) =>
+        {
+            HeadersDirect.Reset();
+
+            var response = context.Response;
+            response.StatusCode = 200;
+            response.ContentType = "text/plain";
+            response.Headers["Content-Length"] = "13";
+
+            response.Headers["Unknown"] = "Unknown";
+            response.Headers["IUnknown"] = "Unknown";
+            response.Headers["X-Unknown"] = "Unknown";
+
+            var dateHeaderValues = DateHeaderValueManager.GetDateHeaderValues();
+            HeadersDirect.SetRawDate(dateHeaderValues.String, dateHeaderValues.Bytes);
+            return Task.CompletedTask;
+        };
+
+        private static readonly RequestDelegate ContentLengthString = (context) =>
+        {
+            var response = context.Response;
+            response.Headers["Content-Length"] = "13";
+            return Task.CompletedTask;
+        };
+
+        private static readonly RequestDelegate ContentLengthNumeric = (context) =>
+        {
+            var response = context.Response;
+            response.ContentLength = 13;
+            return Task.CompletedTask;
+        };
+
+        [Setup]
+        public void Setup()
+        {
+            var trace = new KestrelTrace(new TestKestrelTrace());
+            var threadPool = new LoggingThreadPool(trace);
+
+            MemoryPool = new MemoryPool();
+            SocketInput = new SocketInput(MemoryPool, threadPool);
+
+            var connectionContext = new MockConnection(new KestrelServerOptions());
+            connectionContext.Input = SocketInput;
+
+            var httpContextFactory = new HttpContextFactory(new DefaultObjectPoolProvider(), Options.Create(new FormOptions()));
+
+            var testType = GetRequestDelegate(Type);
+
+            Application = new HostingApplication(testType, new NullScopeLogger(), new NoopDiagnosticSource(), httpContextFactory);
+
+            var frame = new Frame<HostingApplication.Context>(application: Application, context: connectionContext);
+            frame.Reset();
+            frame.InitializeHeaders();
+            HeadersDirect = (FrameResponseHeaders)frame.ResponseHeaders;
+            Context = Application.CreateContext(frame);
+        }
+
+        [Cleanup]
+        public void Cleanup()
+        {
+            Application.DisposeContext(Context, null);
+            SocketInput.IncomingFin();
+            SocketInput.Dispose();
+            MemoryPool.Dispose();
+        }
+
+        private class NullScopeLogger : ILogger
```

Comment:
Can't you use this https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Abstractions/NullLogger.cs instead?